backend/tests/controllers/test_cases/test_reported_issues.py

```diff
@@ -0,0 +1,145 @@
+from collections.abc import Callable
+from typing import Any, TypeAlias
+
+import pytest
+from fastapi.testclient import TestClient
+from httpx import Response
+
+endpoint = "/v1/test-cases/reported-issues"
+valid_post_data = {
+    "template_id": "template 1",
+    "url": "http://issue.link/",
+    "description": "some description",
+}
+
+
+@pytest.fixture
+def post(test_client: TestClient):
+    def post_helper(data: Any) -> Response:  # noqa: ANN401
+        return test_client.post(endpoint, json=data)
+
+    return post_helper
+
+
+@pytest.fixture
+def get(test_client: TestClient):
+    def get_helper(query_parameters: dict[str, str] | None = None) -> Response:
+        return test_client.get(endpoint, params=query_parameters)
+
+    return get_helper
+
+
+@pytest.fixture
+def put(test_client: TestClient):
+    def put_helper(id: int, data: Any) -> Response:  # noqa: ANN401
+        return test_client.put(f"{endpoint}/{id}", json=data)
+
+    return put_helper
+
+
+@pytest.fixture
+def delete(test_client: TestClient):
+    def delete_helper(id: int) -> Response:
+        return test_client.delete(f"{endpoint}/{id}")
+
+    return delete_helper
+
+
+Post: TypeAlias = Callable[[Any], Response]
+Put: TypeAlias = Callable[[int, Any], Response]
+Get: TypeAlias = Callable[..., Response]
+Delete: TypeAlias = Callable[[int], Response]
```

Comment:
We seem to have the same fixtures defined in `tests/controllers/test_executions/test_reruns.py`. Perhaps we can move them to a shared file and use it in both, maybe in `tests/common.py` or `test/fixtures.py`?

Reply:
Code wise they are similar. But they represent different pieces of knowledge. These fixtures know how to get, put, delete and post on this particular endpoint. But they don't necessarily work the same way for other endpoints. For reruns for instance, `delete` works differently (it takes a body that contains a list of ids to delete). For test executions (start_test, end_test, etc), put works differently as in it doesn't require an id on path. Admittedly, this shows that TO API is not very uniform. This is something that we should take into consideration when migrating to v2. But as is, it would be difficult to reuse such fixtures across endpoints.

Reply:
Frankly, while these fixtures do reduce some common code, it's probably arguable whether they're worth it. As in, I wouldn't mind removing them if that's easier to read.

Reply:
Oh, I see, I didn't notice the slight differences between the API endpoints we have. I agree about making them more uniform when we migrate to a newer version.